app/models/user.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable,
+         :omniauthable, omniauth_providers: [:google_oauth2]
+
+  def self.from_omniauth(auth)
+    find_existing_user(auth) || create_new_user(auth)
```

Comment:
ActiveRecord has a method `find_or_initialize_by` that can remove the OR block 💡

Reply:
@olivierobert Since we will need to create a new record and save it to db if there is no existing user, I will update to use `find_or_create_by` instead ([find_or_initialize_by](https://guides.rubyonrails.org/active_record_querying.html#find-or-initialize-by) just create new record in memory) 😄